code/modules/power/supermatter/supermatter.dm

```diff
@@ -736,50 +739,51 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		if(damage > explosion_point)
 			countdown()
 
-	return 1
+	return TRUE
 
-/obj/machinery/power/supermatter_crystal/bullet_act(obj/projectile/Proj)
-	var/turf/L = loc
-	if(!istype(L))
+/obj/machinery/power/supermatter_crystal/bullet_act(obj/projectile/projectile)
+	var/turf/local_turf = loc
+	if(!istype(local_turf))
 		return FALSE
-	if(!istype(Proj.firer, /obj/machinery/power/emitter) && power_changes)
-		investigate_log("has been hit by [Proj] fired by [key_name(Proj.firer)]", INVESTIGATE_SUPERMATTER)
-	if(Proj.flag != BULLET)
+	if(!istype(projectile.firer, /obj/machinery/power/emitter) && power_changes)
+		investigate_log("has been hit by [projectile] fired by [key_name(projectile.firer)]", INVESTIGATE_SUPERMATTER)
+	if(projectile.flag != BULLET)
```

Comment:
Aren't these bitfields? Shouldn't this check `&`? I know you didn't write this, but double check for me.

Reply:
is not a bitfield but is a normal datum `#define BULLET "bullet"`
no idea why is called flag in the projectile thing and not something like type since they define the type of the projectile (bullet, energy, etc)